test/units/purge_test.rb

```diff
@@ -1,31 +1,31 @@
 # frozen_string_literal: true
 
 require "test_helper"
+require "json"
 
 class PurgeTest < Imgix::Test
   def test_runtime_error_without_api_key
     assert_raises(RuntimeError) do
-      mock_client(api_key: nil).purge(mock_image)
-    end
-  end
-
-  def test_purger_version_warns
-    stub_request(:post, endpoint).with(body: body).to_return(status: 200)
-
-    assert_output(nil, deprecation_warning) do
-      mock_client(api_key: "10adc394").purge("/images/demo.png")
+      mock_client(api_key: nil).purge(mock_image_url)
     end
   end
 
   def test_successful_purge
-    stub_request(:post, endpoint).with(body: body).to_return(status: 200)
+    stub_request(:post, endpoint).with(
+      body: json_request_body(mock_image_url),
+      headers: mock_headers
+    ).to_return(
+      status: 200,
+      body: "",
+      headers: {}
+    )
 
-    mock_client(api_key: "10adc394").purge("/images/demo.png")
+    mock_client(api_key: mock_api_key).purge("/images/demo.png")
```

Comment:
I think it would be a good idea to add a test here for a path that doesn't start with `/`, to test the branching behaviour above.

Reply:
That'd be great! Let's do that.